pkg/provisioner/controller.go

```diff
@@ -293,24 +300,34 @@ func (c *Controller) handleProvisionClaim(key string, obc *v1alpha1.ObjectBucket
 		return fmt.Errorf("error %s bucket: %v", verb, err)
 	} else if ob == (&v1alpha1.ObjectBucket{}) {
 		return fmt.Errorf("provisioner returned nil/empty object bucket")
+	} else if err := validateObjectBucket(ob); err != nil {
```

Comment:
should be`=` instead of `:=` if we want the deferred delete-on-error to run in this case or else the outer err will stay nil